core/git_command.py

```diff
@@ -31,6 +31,22 @@ class GitSavvyError(Exception):
     pass
 
 
+class IsolatedGitCommand(GitCommand):
```

Comment:
Yeah it's a little different, but I needed git outside of a command/window so I didn't need the rest of the baggage... really what should be abstracted is running commands in general then git on top of that, then GitCommand as used by window/view.